frontend/src/metabase/visualizations/components/TableInteractive/TableInteractive.jsx

```diff
@@ -879,7 +907,13 @@ class TableInteractive extends Component {
         }}
       >
         <Box
-          ref={e => (this.headerRefs[columnIndex] = e)}
+          ref={element => {
+            // We cannot have `null` in `nodeRef` as it will fallback to
+            // `findDOMNode` which no longer exists in React 19.
+            if (element && this.headerRefs[columnIndex]) {
+              this.headerRefs[columnIndex].current = element;
+            }
+          }}
```

Comment:
Without the `this.headerRefs[columnIndex]` check, some e2e tests will fail.

Seems like in some edge cases, the first render doesn't have `headerRefs` initialized yet for some reason despite it being in `componentDidMount` - must be due to React lifecycles